File: JustBaiDuMapLocationDemo/app/src/main/java/com/example/maplocationdemo/MainActivity.java
package com.example.maplocationdemo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.maplocationdemo.baidu.location.BaiDuLocationActivity;
import com.example.maplocationdemo.baidu.map.BaiDuMapActivity;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btBaiduLocation = findViewById(R.id.bt_baidu_location);
        btBaiduLocation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, BaiDuLocationActivity.class);
                startActivity(intent);
            }
        });

        Button btBaiduMap = findViewById(R.id.bt_baidu_map);
        btBaiduMap.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, BaiDuMapActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: KotlinMessager/app/src/main/java/com/example/acer/myapplication/MainActivity.kt
package com.example.acer.myapplication

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

import android.util.Log
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        register_button_register.setOnClickListener(){
            performRegister()
        }
        already_have_a_count_textview_register.setOnClickListener(){
            Log.d("MainActivity","try to show Login activity")

            val intent = Intent(this,LoginActivity::class.java)
            startActivity(intent)
        }
        
    }
    private fun performRegister(){
        val email = email_edittext_register.text.toString()
        val password = password_edittext_register.text.toString()

        if(email.isEmpty() || password.isEmpty()){
            Toast.makeText(this,"请输入用户名密码",Toast.LENGTH_SHORT).show()
            return
        }

        Log.d("MainActivity","Email is:"+email)
        Log.d("MainActivity","Password is:$password")

        FirebaseAuth.getInstance().createUserWithEmailAndPassword(email,password)
            .addOnCompleteListener{
                if (!it.isSuccessful)
                    return@addOnCompleteListener
                //else if successful
                Log.d("Main","成功创建一个用户，该用户的uid是：${it.result.user.uid}")
            }
            .addOnFailureListener{
                Log.d("Main","创建用户失败：${it.message}")
                Toast.makeText(this,"创建用户失败：${it.message}",Toast.LENGTH_SHORT).show()
            }
    }
}
